elements-fun/src/transaction.rs

```diff
@@ -469,37 +1006,37 @@ impl TxOut {
 
     /// Whether or not this output is a fee output
     pub fn is_fee(&self) -> bool {
-        self.script_pubkey.is_empty() && self.value.is_explicit() && self.asset.is_explicit()
+        self.script_pubkey().is_empty() && matches!(self, Self::Explicit(_))
     }
 
     /// Extracts the minimum value from the rangeproof, if there is one, or returns 0.
     pub fn minimum_value(&self) -> u64 {
-        let min_value = if self.script_pubkey.is_op_return() {
+        let min_value = if self.script_pubkey().is_op_return() {
             0
         } else {
             1
         };
 
-        match self.value {
-            confidential::Value::Null => min_value,
-            confidential::Value::Explicit(n) => n,
-            confidential::Value::Confidential(..) => {
-                if self.witness.rangeproof.is_empty() {
+        match self {
+            Self::Null { .. } => min_value,
+            Self::Explicit(explicit) => explicit.value.0,
+            Self::Confidential(ConfidentialTxOut { witness, .. }) => {
+                if witness.rangeproof.is_empty() {
                     min_value
                 } else {
-                    debug_assert!(self.witness.rangeproof.len() > 10);
+                    debug_assert!(witness.rangeproof.len() > 10);
 
-                    let has_nonzero_range = self.witness.rangeproof[0] & 64 == 64;
-                    let has_min = self.witness.rangeproof[0] & 32 == 32;
+                    let has_nonzero_range = witness.rangeproof[0] & 64 == 64;
+                    let has_min = witness.rangeproof[0] & 32 == 32;
 
                     if !has_min {
                         min_value
                     } else if has_nonzero_range {
-                        bitcoin::consensus::deserialize::<u64>(&self.witness.rangeproof[2..10])
+                        bitcoin::consensus::deserialize::<u64>(&witness.rangeproof[2..10])
```

Comment:
:thinking: Imagine if we had a rangeproof struct.

Reply:
It is all possible :D